src/components/PeopleTable.tsx

```diff
@@ -1,5 +1,64 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+// /* eslint-disable jsx-a11y/control-has-associated-label */
+import { Person } from '../types/Person';
+import { Link, useParams, useSearchParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { SearchLink } from '../components/SearchLink';
+import cn from 'classnames';
+import { useCallback } from 'react';
+
+interface PeopleTableProps {
+  persons: Person[];
+
+  sortField: string | null;
+  sortOrder: 'asc' | 'desc' | null;
+}
+
+export const PeopleTable: React.FC<PeopleTableProps> = ({
+  persons,
+  sortField,
+  sortOrder,
+}) => {
+  const findPersonByName = (name: string | null) => {
+    return persons.find(p => p.name === name);
+  };
+
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [searchParams] = useSearchParams();
+  const { slug } = useParams();
+
+  useEffect(() => {
+    const storedSlug = searchParams.get('selectedPerson');
+
+    if (storedSlug) {
+      const selectPerson = persons.find(person => person.slug === storedSlug);
+
+      setSelectedPerson(selectPerson || null);
+    }
+
+    if (slug) {
+      const selectedPersonSlug = persons.find(person => person.slug === slug);
+
+      setSelectedPerson(selectedPersonSlug || null);
+    }
+  }, [persons, searchParams, slug]);
+
+  const getSortParams = useCallback(
+    (sortBy: string) => {
+      if (sortBy === sortField) {
+        if (!sortOrder) {
+          return { sort: sortBy, order: 'desc' };
+        }
+
+        if (sortOrder === 'desc') {
+          return { sort: null, order: null };
+        }
+      }
+
+      return { sort: sortBy, order: null };
```

Comment:
The return statement here should set the order to 'asc' instead of null when the sortBy is different from the current sortField. This ensures that the sorting starts in ascending order when a new field is selected.